### PR TITLE
25 refactor pages

### PR DESCRIPTION
**Pull request by HatemMn**
Goal :
- This PR was created to centralize the paths of the webpage inside one new file : routePathsConfig.tsx and remove the need to write ids manually or change many many files to include a new page
- And also to update some dependencies

How it was done :
The new functions within navigationConfig.tsx transforms that centralized file into the legacy dataStructures ( NavigationItem and ComponentsList ), which are then imported

What will change : 
This PR should **not** introduce a breaking change (or any change in the first place) with how the rest of the project works.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (HatemMn) [inline CONTRIBUTING.md:1]: by the way i didn't know what was the convention we use is so I added this file to separate the instructions of contribution to the repo from those to use it ( the README file )

File: public/actions/javascript/summarizeDocumentContent.ts
Comment (HatemMn) [inline public/actions/javascript/summarizeDocumentContent.ts:1]: as you can see here : https://github.com/Cosmian/interactive-tutorial-ui/blob/main/src/actions/javascript/summarizeDocumentContent.ts

my file is identical to the one on the main branch - why is github marking a diff ? I spent half an hour looking and do not it figure out why this is showing up, despite it not really causing trouble

any idea @ccorsin ?

File: src/utils/navigationConfig.tsx [hunk 1]
```diff
@@ -1,225 +1,79 @@
...
+  const specialCases: Record<string, string> = {
+    "client-side-encryption": "Encrypt Client-Side",
+  };
+  if (specialCases[key]) return specialCases[key];
+  return key
+    .replace(/-/g, " ")
+    .replace(/\b\w/g, (c) => c.toUpperCase())
+    .replace(/vm/g, "VM")
+    .replace(/cosmian/g, "Cosmian");
+}
```
Comment (ccorsin) [inline src/utils/navigationConfig.tsx:77]: Why don't we simply make label mandatory - and remove this function ? This seems a little over-engineered and could be handle defining a type for navigation content

File: src/utils/routePathsConfig.tsx [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+import GetCertificate from "../pages/pki/GetCertificate";
+import GrantAccess from "../pages/pki/GrantAccess";
+import ImportAndUnwrapUDK from "../pages/pki/ImportAndUnwrapUDK";
+import SendWrappedDecryptionKey from "../pages/pki/SendWrappedDecryptionKey";
+import SetupPki from "../pages/pki/SetupPki";
+import UploadCert from "../pages/pki/UploadCert";
+
+export interface routePath {
+  path: string;
+  label?: string;
```
Comment (ccorsin) [inline src/utils/routePathsConfig.tsx:35]: make this mandatory

File: src/utils/routePathsConfig.tsx [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+  component?: JSX.Element;
```
Comment (ccorsin) [inline src/utils/routePathsConfig.tsx:36]: make this mandatory ?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+## Adding new pages :
+
+The single page tree structure is centralized in the following file : [`src/utils/pageConfig.tsx`](src/utils/pageConfig.tsx).
```
Comment (ccorsin) [inline CONTRIBUTING.md:3]: I think the link to the file is incorrect now - routePathsConfig

**Comment by HatemMn** [comment]
Will close this PR and push a renamed branch with more easy to read names, sorry for the inconvenience

File: src/utils/navigationConfig.tsx (hunk 1 above)
Comment (HatemMn) [reply src/utils/navigationConfig.tsx:78]: the purpose of this function has been redefined and now it serves only as fallback, what do you think ? We can still delete but we should make label mandatory and hence more checks

The idea was to have the least possible "mandatory" fields in our object to make updating the codebase easier later on if someone has yet another thing to add on top of what we already have

File: CONTRIBUTING.md (hunk 4 above)
Comment (HatemMn) [reply CONTRIBUTING.md:3]: will be corrected thx

**Pull request closed without merging**